backfill/2022-08-31-decoder-failure/README.md

```diff
@@ -0,0 +1,151 @@
+
+See the script in this directory for the invocation to run the Decoder backfill
+job to populate the staging live tables.
+
+With a local working copy of `gcp-ingestion` in sync with `main`, run the [`02-gcp-ingestion-beam-decoder.sh` script](02-gcp-ingestion-beam-decoder.sh) to start the decoder as a Dataflow job:
+
+```bash
+cd path/to/gcp-ingestion
+
+path/to/02-gcp-ingestion-beam-decoder.sh
+```
+
+## OPTIONAL: Set up a GCP instance in the backfill project to run API calls from
+
+This is advisable if you're running a lot of API calls from an unreliable
+network, such as conference WiFi.
+
+This can be done from the GCP web console if you're an Editor on the project:
+make sure to use the default service account and enable all scopes so that the
+instance can make BQ API calls.
+
+```bash
+gcloud compute instances create dsre-999-backfill --project=moz-fx-data-backfill-10 --zone=us-west1-b --machine-type=e2-standard-4 --network-interface=subnet=default,no-address --metadata=enable-oslogin=true --maintenance-policy=MIGRATE --service-account=339436751542-compute@developer.gserviceaccount.com --scopes=https://www.googleapis.com/auth/cloud-platform --create-disk=auto-delete=yes,boot=yes,device-name=dsre-999-backfill,image=projects/debian-cloud/global/images/debian-11-bullseye-v20220920,mode=rw,size=10,type=projects/moz-fx-data-backfill-10/zones/us-central1-a/diskTypes/pd-balanced --no-shielded-secure-boot --shielded-vtpm --shielded-integrity-monitoring --reservation-affinity=any
+```
+
+To SSH into the instance once it has been created, run:
+
+```
+gcloud compute ssh --zone "us-west1-b" "dsre-999-backfill"  --tunnel-through-iap --project "moz-fx-data-backfill-10"
+```
+
+## Step 3: Copy & deduplicate decoded pings
+
+1. Determine which staging live tables need to be processed and update the script in this directory
+
+```bash
+bq query --format csv --nouse_legacy_sql 'SELECT FORMAT("%s.%s", table_schema, table_name) as table_id FROM `moz-fx-data-backfill-10.region-us.INFORMATION_SCHEMA.TABLE_STORAGE_BY_PROJECT` WHERE creation_time > "2022-08-31" and table_schema LIKE "%_live" and total_rows > 0;'
+```
+
+Note: the `creation_time` filter shouldn't strictly be necessary but
+information schema metadata tables appear to contain possibly stale metadata
+from deleted tables such as from the [last backfill that was run from this project](https://github.com/mozilla/bigquery-backfill/pull/18).
+
+2. Run the script in this directory for the invocation of relevant `copy_deduplicate` queries:
+
+```bash
+cd path/to/bigquery-etl
+
+./path/to/03-bigquery-etl-copy-deduplicate.sh
+```
+
+## Step 4: Insert previously missing pings into production tables
+
+Because of the way the initial backfill query was structured, we should be guaranteed that documents we're processing in this backfill don't correspond to an existing document in the stable tables, and so a direct append of affected records should be acceptable.
```

Comment:
This needs a fact check.

Reply:
@relud can you verify this statement?

Reply:
yes, the query earlier in this document only staged missing documents